docs/fsharp-cheatsheet.md

```diff
@@ -169,197 +205,281 @@ The `rec` keyword is used together with the `let` keyword to define a recursive
         | { Name = "Paul" } -> true
         | _ -> false
 
-<div id="discriminated-unions"></div>
+See [Records (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/records) to learn more; including `struct`-based records.
+
+<div id="data-types-anonymous-records"></div>
+
+### Anonymous Records
+
+*Anonymous Records* represent aggregates of named values, but do not need declaring before use.
+
+    // Create
+    let anonRecord1 = {| Name = "Don Syme"; Language = "F#"; Age = 999 |}
 
-## Discriminated Unions
+    // Duplicate
+    let anonRecord2 = {| anonRecord1 with Name = "Mads Torgersen"; Language = "C#" |}
+
+    let getCircleStats (radius:float) =
+        {| Radius = radius
+           Diameter = radius * 2.0
+           Area = System.Math.PI * (radius ** 2.0)
+           Circumference = 2.0 * System.Math.PI * radius |}
+
+    // Signature
+    let printCircleStats (circle: {| Radius: float; Area: float; Circumference: float; Diameter: float |}) =
+        printfn $"Circle with R=%f{circle.Radius}; D=%f{circle.Diameter}; A=%f{circle.Area}; C=%f{circle.Circumference}"
+
+    let cc = getCircleStats 2.0
+    printCircleStats cc
+
+See [Anonymous Records (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/anonymous-records) to learn more; including `struct`-based anonymous records.
+
+<div id="data-types-discriminated-unions"></div>
+
+### Discriminated Unions
 
 *Discriminated unions* (DU) provide support for values that can be one of a number of named cases, each possibly with different values and types.
 
+    // Declaration
+    type Interaction =
+        | Keyboard of char
+        | KeyboardWithModifier of char * modifier:System.ConsoleModifiers
+        | MouseClick of countOfClicks:int
+
+    // Create
+    let interaction1 = MouseClick 1
+    let interaction2 = MouseClick (countOfClicks = 2)
+    let interaction3 = KeyboardWithModifier ('c', System.ConsoleModifiers.Control)
+
+    // Pattern matching
+    match interaction3 with
+    | Keyboard chr -> $"Character: {chr}"
+    | KeyboardWithModifier (chr, modifier) -> $"Character: {modifier}+{chr}"
+    | MouseClick (countOfClicks = 1) -> "Click"
+    | MouseClick (countOfClicks = x) -> $"Clicked: {x}"
+
+Generics
+
     type Tree<'T> =
         | Node of Tree<'T> * 'T * Tree<'T>
         | Leaf
 
-    let rec depth = function
-        | Node(l, _, r) -> 1 + max (depth l) (depth r)
+    let rec depth =
+        match depth with
+        | Node (l, _, r) -> 1 + max (depth l) (depth r)
         | Leaf -> 0
 
 F# Core has built-in discriminated unions for error handling, e.g., [`option`](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/options) and [`Result`](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/results).
 
     let optionPatternMatch input =
         match input with
-        | Some i -> printfn "input is an int=%d" i
+        | Some value -> printfn $"input is {value}"
```

Comment:
see earlier comment about `f` vs interpolation
also I think missing omitting `%d` is good here but is the difference explained anywhere?